File: src/Domain/Entities/Baja.cs
using TrazabilidadIberica.Domain.Common;
using TrazabilidadIberica.Domain.Enums;

namespace TrazabilidadIberica.Domain.Entities;

public class Baja : BaseEntity
{
    public Guid AnimalId { get; set; }
    public DateTime FechaBaja { get; set; }
    public CausaBaja Causa { get; set; }
    public string? Destino { get; set; }
    public string? NumGuiaAsociada { get; set; }
    public string? Observaciones { get; set; }

    public Animal Animal { get; set; } = null!;
}

File: frontend/src/app/core/models/baja.ts
export enum CausaBaja {
  Venta = 0,
  Muerte = 1,
  Sacrificio = 2,
  Perdida = 3,
  Donacion = 4,
}

export interface Baja {
  id: string;
  animalId: string;
  fechaBaja: string;
  causa: CausaBaja;
  destino: string | null;
  numGuiaAsociada: string | null;
  observaciones: string | null;
  animal?: { id: string; numeroCrotal: string };
  createdOffline: boolean;
  clientId: string | null;
  createdAt: string;
  updatedAt: string;
  deletedAt: string | null;
}
